### PR TITLE
Mohammadsanaee patch 1

### PR DESCRIPTION
**Pull request by mohammadsanaee**

### CODE REVIEW

**Comment by mohammadsanaee** [comment]
Please review the readme file
Thanks

File: reverse_image_search_module/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+
+# Image Retrieval using ResNet-18 and Annoy Index
+
+This folder contains code for performing reverse image search using the ResNet-18 neural network for feature extraction and the Annoy Index for efficient nearest neighbor search. The provided code allows you to generate embeddings for a collection of images, build an Annoy Index from those embeddings, and perform reverse image searches to find similar images from a dataset.
+
+
+# Image Retrieval using ResNet-18 and Annoy Index
+
+This repository contains code for performing reverse image search using the ResNet-18 neural network for feature extraction and the Annoy Index for efficient nearest neighbor search.
```
Comment (aguschin) [inline reverse_image_search_module/README.md:9]: duplicates, but otherwise looks good!

**Review by aguschin — COMMENTED** [review]
LGTM!

What's missing is "possible future steps" that will explain how you can improve quality, speed, or something else. You can discuss trying different models instead of resnet to get better quality of search, ideas on writing automated tests to validate your changes, ways to measure quality of your search, etc.

File: reverse_image_search_module/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+img2vec = Img2VecResnet18()
+
+# Load an image and get its normalized embedding
+image = plt.imread('path_to_your_image.jpg')
+normalized_embedding = img2vec.getNormalizedVec(image)
+```
+
+## Building the Annoy Index
+
+The Annoy Index is built using embeddings from a collection of images.
```
Comment (aguschin) [inline reverse_image_search_module/README.md:55]: You can say couple of words about why annoy (it's quite efficient and easy to use - maybe even easier than sklearn or on the same level)

File: reverse_image_search_module/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+from resnet_image_search import find_image
+
+# Load the image you want to search for
+query_image = plt.imread('path_to_query_image.jpg')
+
+# Perform reverse image search
+idx, dist, data = find_image(query_image)
+
+# Display the search results
+print(f"Most similar image: {data['title']} (Distance: {dist:.2f})")
```
Comment (aguschin) [inline reverse_image_search_module/README.md:99]: I think you need to mention part about checking the distance to decide it's not the right finding. It's part of telegram bot logic though, so also mention that. Maybe mention selecting this parameter optimally as a future step to improve this)

**Comment by mohammadsanaee** [comment]
Thanks for your time and comments🙏🏼
I will complete it

**Review by aguschin — APPROVED** [review]
I'm going to approve in case you want to merge this!

**Review by abuzarmd-ML — APPROVED** [review]
can you write future work-

Future Work: We will replace the numpy file where we are storing embeddings, with a normal database, we are planning to replace the file with postgresSql.

**Pull request merged**